Source/Extensions/Blazorise.Components/ListView.razor

```diff
@@ -1,20 +1,36 @@
-﻿@typeparam TItem
+﻿@using Microsoft.AspNetCore.Components.Web.Virtualization
+@typeparam TItem
 <ListGroup Class="@ListGroupClassNames" Style="@ListGroupStyleNames" Flush="@Flush" Scrollable="@Scrollable" Mode="@Mode" SelectedItem="@GetItemValue( SelectedItem )" SelectedItemChanged="@SelectedListGroupItemChanged" Attributes="@Attributes">
     @if ( Data != null )
     {
-        @foreach ( var item in Data ?? Enumerable.Empty<TItem>() )
+        @if ( Virtualize && Data is ICollection<TItem> collectionableData )
         {
-            var itemText = GetItemText( item );
-            var itemValue = GetItemValue( item );
-            @if ( ItemTemplate is not null )
-            {
-                <ListGroupItem @key="item" Name="@itemValue">@ItemTemplate( new( item, itemText, itemValue ) )</ListGroupItem>
-            }
-            else
+            <Virtualize TItem="TItem" Context="item" Items="@collectionableData">
+                @itemFragment(item)
+            </Virtualize>
+        }
+        else
+        {
+            @foreach ( var item in Data ?? Enumerable.Empty<TItem>() )
             {
-                <ListGroupItem @key="item" Name="@itemValue">@itemText</ListGroupItem>
+                @itemFragment(item)
             }
         }
     }
     @ChildContent
 </ListGroup>
+@code {
+    protected RenderFragment<TItem> itemFragment => item => __builder =>
+    {
+        var itemText = GetItemText( item );
+        var itemValue = GetItemValue( item );
+        @if ( ItemTemplate is not null )
+        {
+            <ListGroupItem @key="item" Name="@itemValue">@ItemTemplate( new( item, itemText, itemValue ) )</ListGroupItem>
+        }
+        else
+        {
+            <ListGroupItem @key="item" Name="@itemValue">@itemText</ListGroupItem>
```

Comment:
If we think / assume, itemvalue will always be unique, we could change the key to that, we did assume that in SelectList apparently.

Reply:
The user should handle the uniqueness of data.